### PR TITLE
Sanitizing parameters

### PR DESCRIPTION
**Pull request by ds-ms**

### CODE REVIEW

File: Tasks/AzureResourceGroupDeployment/task.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
     "version": {
         "Major": 2,
         "Minor": 129,
```
Comment (sachinma) [inline Tasks/AzureResourceGroupDeployment/task.json]: Should it be 130 now; or you plan to port this to 129

File: Tasks/AzureResourceGroupDeployment/operations/ResourceGroup.ts [hunk 2]
```diff
@@ -366,6 +366,8 @@ export class ResourceGroup {
             parameters = this.updateOverrideParameters(template, parameters);
         }
 
+        parameters = this.sanitizeParameters(parameters);
```
Comment (sachinma) [inline Tasks/AzureResourceGroupDeployment/operations/ResourceGroup.ts]: This needs to be called in PublicURL deployment as well

File: Tasks/AzureResourceGroupDeployment/task.json (hunk 1 above)
Comment (sachinma) [reply Tasks/AzureResourceGroupDeployment/task.json]: My recommendation: Lets port this to 129

**Comment by sachinma** [comment]
Also add some UTs

File: Tasks/AzureResourceGroupDeployment/operations/ResourceGroup.ts (hunk 2 above)
Comment (ds-ms) [reply Tasks/AzureResourceGroupDeployment/operations/ResourceGroup.ts]: done

File: Tasks/AzureResourceGroupDeployment/task.json (hunk 1 above)
Comment (ds-ms) [reply Tasks/AzureResourceGroupDeployment/task.json]: I'll cherry pick to 129

**Comment by boevink** [comment]
Sanitizing parameters seems to fail voor parameters of type bool:
[issue 6339](https://github.com/Microsoft/vsts-tasks/issues/6339)

**Pull request merged**